gravity.sh

```diff
@@ -150,34 +166,40 @@ database_table_from_file() {
 # Migrate pre-v5.0 list files to database-based Pi-hole versions
 migrate_to_database() {
   # Create database file only if not present
-  if [ -e "${gravityDBfile}" ]; then
-    return 0
+  if [ ! -e "${gravityDBfile}" ]; then
+    echo -e "  ${INFO} Creating new gravity database"
+    generate_gravity_database
   fi
 
-  echo -e "  ${INFO} Creating new gravity database"
-  generate_gravity_database
+  # Check if gravity database needs to be updated
+  upgrade_gravityDB "${gravityDBfile}"
 
   # Migrate list files to new database
```

Comment:
These migrations should only be called once. For this audit log migration, the code to migrate the audit log should be contained as part of that specific migration.

Reply:
They will check against the version and immediately exit. It seems to be the best solution. When we would make it dependent on the existence of `audtlog.list`, then the migrations would never be triggered for most users.

Reply:
Using the example of the audit list, the only case when we want to run the migration is when we are upgrading the database from 1 to 2. Therefore, we should only run the migration right after we upgrade the database to version 2. If we always run the migration (when the file is there), then we are introducing a lot of uncertainty into the upgrade process. Instead of being in one place, the migration happens due to a combination of separate events.